README.pod

```diff
@@ -98,40 +96,171 @@ C<dispose> method:
+  my $query = 'SELECT id FROM tab WHERE id IN (%s)'.sprintf($parameter-bind-marks);
+  $dbh.execute($query, |@value-list);
+
+All database drivers support basic Raku types like Int, Rat, Str, and Buf; some databases may support additional
+complex types such as an Array of Str or Array of Int which may simplify the above example significantly. Please see
+database specific documentation for descriptions of additional type support.
+
+=head3 C<prepare>
+
+Execute performs a couple of steps on the client side, and often the database side as well, which may be cached
+if a query is going to be executed several times. For simple queries, prepare() may increase performance by up to 50%
+
+  # Inefficient
+  for 1 .. 100 -> $id {
+    $dbh.execute('INSERT INTO tab (id) VALUES (?)', $id);
+  }
 
-DBIish provides nearly all the perl5 DBI fetch* method to fetch values from the C<StatementHandle> object.
-However it's recommended to use the C<row> and C<allrows> methods. They provide you typed values
+  # Efficient
+  #
+  # Uses prepare to decrease overhead on the client side; often on the server-side too as the database may only need
+  # to parse the SQL once.
+  my $sth = $dbh.prepare('INSERT INTO tab (id) VALUES (?)');
+  for 1 .. 100 -> $id {
+    $sth.execute($id);
+  }
 
-=head3 row
+=head2 Retrieving data
+
+DBIish provides the C<row> and C<allrows> methods to fetch values from a C<StatementHandle> object returned by execute.
+These functions provide you typed values; for example an int4 field in the database will be provided as an Integer
+ scalar in Raku.
+
+=head3 C<row>
 
 C<row> take the C<hash> adverb if you want to have the values in a Hash form instead of a plain Array
 
 Example:
 
+    my $sth = $dbh.execute('SELECT id, col FROM tab WHERE id = ?', $value-id);
+
     my @values = $sth.row();
     my %values = $sth.row(:hash);
 
-=head3 allrows
+=head3 C<allrows>
 
 C<allrows> lazily returns all the rows as a list of arrays.
 If you want to fetch the values in a hash form, use one of the two adverbs C<array-of-hash>
 or C<hash-of-array>
 
 Example:
 
+    my $sth = $dbh.execute('SELECT id, col FROM tab');
+
     my @data = $sth.allrows(); # [[1, 'val1'], [3, 'val2']]
-    my @data = $sth.allrows(:array-of-hash); # [ ( a => 1, b => 'val1'), ( a => 3, b => 'val2') ]
-    my %data = $sth.allrows(:hash-of-array); # a => [1, 3], b => ['val1', 'val2']
+    my @data = $sth.allrows(:array-of-hash); # [ ( id => 1, col => 'val1'), ( id => 3, col => 'val2') ]
+    my %data = $sth.allrows(:hash-of-array); # id => [1, 3], col => ['val1', 'val2']
+
+    for $sth.allrows(:array-of-hash) -> $row {
+      say $row<id>;  # 1␤3
+    }
+
+    # Or as a shorter example:
+    for $dbh.execute('SELECT id, col FROM tab').allrows(:array-of-hash) -> $row {
+       say $row<id>  # 1␤3
+    }
+
+=head3 C<dispose>
+
+After you have finished with the statement handle, you can free the memory it used immediately using C<dispose>.
```

Comment:
After you have fetched all data using the statement handle, you can free its memory immediately using C<dispose>.